From e4d9b1869933051549829ba34ba3dd9db6af0c89 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:24:52 -0400
Subject: [tools] perf/sort: Fix a memory leak on srcline

Message-id: <1408462094-14194-124-git-send-email-jolsa@redhat.com>
Patchwork-id: 88157
O-Subject: [PATCH RHEL7.1 BZ1131394 123/325] perf sort: Fix a memory leak on srcline
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 963ba5fd5d04f36d6a5c9a94562484a4f270c1de
Author: Namhyung Kim <namhyung.kim@lge.com>
Date: Wed Sep 11 14:09:25 2013 +0900

description
===========
In the hist_entry__srcline_snprintf(), path and self->srcline are
pointing the same memory region, but they are doubly allocated.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/sort.c b/tools/perf/util/sort.c
index b4ecc0e..97cf3ef 100644
--- a/tools/perf/util/sort.c
+++ b/tools/perf/util/sort.c
@@ -269,10 +269,7 @@ static int hist_entry__srcline_snprintf(struct hist_entry *self, char *bf,
  if (!fp)
   goto out_ip;
 
- if (getline(&path, &line_len, fp) < 0 || !line_len)
-  goto out_ip;
- self->srcline = strdup(path);
- if (self->srcline == NULL)
+ if (getline(&self->srcline, &line_len, fp) < 0 || !line_len)
   goto out_ip;
 
  nl = strchr(self->srcline, '\n');
-- 
1.7.1